src/Materials/effect.ts

```diff
@@ -669,6 +683,65 @@ export class Effect implements IDisposable {
         }
     }
 
+    public getPipelineLayout(): GPUPipelineLayout {

```

Comment:
This should not be moved in effect as the goal is keep any WebGPU specifics in WebGPUEngine only. People using WebGL, Native or Null engine should not have this code in their distrib.

I guess it would also prevent the GpuDevice exposed ?